roxctl/common/flags/env_flag_test.go

```diff
@@ -30,6 +31,145 @@ func TestFlagOrSettingValue(t *testing.T) {
+	assert.Equal(t, "some-other-other-test-value", caCertFile)
+
+}
+
+func TestFlagOrConfigurationValueEnv(t *testing.T) {
+	var (
+		testFile1 = "./testdata/test_instance1.yaml"
+	)
+
+	cmd := &cobra.Command{
+		PersistentPreRunE: LoadConfig,
+	}
+
+	AddAPITokenFile(cmd)
+	AddConnectionFlags(cmd)
+	AddConfigurationFile(cmd)
+
+	assert.False(t, ConfigurationFileChanged())
+	assert.False(t, *caCertFileSet)
+	assert.Empty(t, *apiTokenFileChanged)
+	assert.False(t, *endpointChanged)
+
+	err := cmd.PersistentFlags().Set("config-file", testFile1)
+	assert.NoError(t, err, "Setting a configuration file should not produce an error")
+	assert.Equal(t, testFile1, ConfigurationFileName())
+
+	// Execute the command to trigger PersistentPreRunE
+	err = cmd.PersistentPreRunE(cmd, []string{})
+	assert.NoError(t, err, "Command execution should not produce an error")
+	assert.True(t, ConfigurationFileChanged())
+	assert.NotEmpty(t, config)
+
+	// 3. Validate that PersistentPreRunE (LoadConfig) ran successfully
+	assert.NotNil(t, config, "Config should be initialized after LoadConfig runs")
+
+	// 4. Change environment values. The configuration-defined flag values should be returned, irrespective if set by environment.
+	t.Setenv("ROX_API_TOKEN_FILE", "some-other-test-env-value")
+	assert.Equal(t, "REDACTED", APITokenFile())
+
+	t.Setenv("ROX_ENDPOINT", "some-test-env-value")
+	assert.Equal(t, "localhost:8000", endpoint)
+
+	t.Setenv("ROX_CA_CERT_FILE", "some-other-other-test-env-value")
+	assert.Equal(t, "./deploy/cert", caCertFile)
+
+}
+
+// TODO: Add flag > config > env setting tests
+func TestFlagOrConfigurationValueFlagsAndEnv(t *testing.T) {
+	var (
+		testFile1 = "./testdata/test_instance1.yaml"
+	)
+
+	cmd := &cobra.Command{
+		PersistentPreRunE: LoadConfig,
+	}
+
+	AddAPITokenFile(cmd)
+	AddConnectionFlags(cmd)
+	AddConfigurationFile(cmd)
+
+	assert.False(t, ConfigurationFileChanged())
+	assert.False(t, *caCertFileSet)
+	assert.Empty(t, *apiTokenFileChanged)
+	assert.False(t, *endpointChanged)
+
+	err := cmd.PersistentFlags().Set("config-file", testFile1)
+	assert.NoError(t, err, "Setting a configuration file should not produce an error")
+	assert.Equal(t, testFile1, ConfigurationFileName())
+
+	// Execute the command to trigger PersistentPreRunE
+	err = cmd.PersistentPreRunE(cmd, []string{})
+	assert.NoError(t, err, "Command execution should not produce an error")
+	assert.True(t, ConfigurationFileChanged())
+	assert.NotEmpty(t, config)
+
+	// 3. Validate that PersistentPreRunE (LoadConfig) ran successfully
+	assert.NotNil(t, config, "Config should be initialized after LoadConfig runs")
+
+	// 4. Change environment values.
+	t.Setenv("ROX_API_TOKEN_FILE", "some-other-test-env-value")
+	t.Setenv("ROX_ENDPOINT", "some-test-env-value")
+	t.Setenv("ROX_CA_CERT_FILE", "some-other-other-test-env-value")
+
+	// 5. Change flag values. Flag values should be returned, irrespective of configuration or environment-defined values.
```

Comment:
I think it would be good to test the precedence of config file settings over env before setting the flags, then to test the precedence of flags over config file settings.